docs/guides/next-cookies.md

```diff
@@ -0,0 +1,232 @@
+
+## Client-side Cookies in Next.js
+
+You set cookies via,
+
+```js
+document.cookie = 'name=value'
+```
+
+You get cookies via,
+
+```js
+document.cookie
+```
+
+Reading `document.cookie` shows all the available cookies in the current context, where a semicolon separates each cookie.
+
+The `document.cookie` property takes up different attributes. E.g. `max-age` lets you define when the cookie should expire, `secure` allows cookies to transmit only over a secure protocol such as HTTPS.
+
+You can read more details about the `document.cookie` property on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie).
+
+### Set and Get Cookies on the Server-side on the Client-side
+
+Let's build a demo application to set a `theme` cookie to save the `light` or `dark` mode of your website.
+
+![client-side-cookies](/docs/guides/images/client-cookies.png)
+
+First create a Next.js page that shows the current value of the cookie`theme` and allows you to change it to `light` or `dark`.
+
+```js
+export default function ClientSideCookiePage() {
+  return (
+    <div>
+      <h1>Client-side Cookies in Next.js</h1>
+      <div>THEME = {theme} </div>
+      <button onClick={() => changeTheme('dark')}>Dark</button>
+      <button onClick={() => changeTheme('light')}>Light</button>
+    </div>
+  )
+}
+```
+
+You can use two React.js hooks to make this app easier to work with:
+
+- [`useState`](https://reactjs.org/docs/hooks-state.html) hook to save the `theme` cookie value
+- [`useEffect`](https://reactjs.org/docs/hooks-effect.html) hook to access the `document`, which is a side-effect
+
+To understand how React hooks work, [check out the React documentation](https://reactjs.org/docs/hooks-intro.html).
+
+Here's the basic structure of our code so far:
+
+```js
+import { useEffect, useState } from 'react'
+
+export default function ClientSideCookiePage() {
+  const [theme, setTheme] = useState()
+
+  useEffect(() => {
+    // Get cookies & find the value of the cookie called `theme`.
+    // Set default cookie to `light` if not already set.
+    // Store the `theme` color in state.
+  })
+
+  // Set theme & cookie to user selected color.
+  const changeTheme = (color) => {}
+
+  return (
+    <div>
+      <h1>Client-side Cookies in Next.js</h1>
+      <div>THEME = {theme} </div>
+      <button onClick={() => changeTheme('dark')}>Dark</button>
+      <button onClick={() => changeTheme('light')}>Light</button>
+    </div>
+  )
+}
+```
+
+And here's the complete code where you get the `theme` cookie and allow the user to set it to `light` or `dark` mode.
+
+```js
+import { useEffect, useState } from 'react'
+
+export default function ClientSideCookiePage() {
+  const [theme, setTheme] = useState()
+
+  useEffect(() => {
+    // Get cookies & find the value of the cookie called `theme`.
+    const color = document.cookie
```

Comment:
docs example must use `decodeURIComponent` on the cookie value before returning.